dtale/global_state.py

```diff
@@ -7,176 +9,298 @@
 
+    @property
+    def dtypes(self):
+        return self._dtypes
 
-def get_dataset(data_id=None):
-    global DATASETS
+    @property
+    def metadata(self):
+        return self._metadata
 
-    if data_id is None:
-        return _as_dict(DATASETS)
-    return DATASETS.get(data_id)
+    @property
+    def context_variables(self):
+        return self._context_variables
+
+    @property
+    def history(self):
+        return self._history
+
+    @property
+    def settings(self):
+        return self._settings
+
+    @property
+    def is_xarray_dataset(self):
+        if self._dataset is not None:
+            return True
+        return False
 
+    @data.setter
+    def data(self, data):
+        self._data = data
 
-def get_dataset_dim(data_id=None):
-    global DATASET_DIM
+    @name.setter
+    def name(self, name):
+        self._name = name
 
-    if data_id is None:
-        return _as_dict(DATASET_DIM)
-    return DATASET_DIM.get(data_id)
+    @dataset.setter
+    def dataset(self, dataset):
+        self._dataset = dataset
 
+    @dataset_dim.setter
+    def dataset_dim(self, dataset_dim):
+        self._dataset_dim = dataset_dim
 
-def get_dtypes(data_id=None):
-    global DTYPES
+    @dtypes.setter
+    def dtypes(self, dtypes):
+        self._dtypes = dtypes
 
-    if data_id is None:
-        return _as_dict(DTYPES)
-    return DTYPES.get(data_id)
+    @context_variables.setter
+    def context_variables(self, context_variables):
+        self._context_variables = context_variables
 
+    @metadata.setter
+    def metadata(self, metadata):
+        self._metadata = metadata
 
-def get_dtype_info(data_id, col):
-    dtypes = get_dtypes(data_id)
-    return next((c for c in dtypes or [] if c["name"] == col), None)
+    @history.setter
+    def history(self, history):
+        self._history = history
 
+    @settings.setter
+    def settings(self, settings):
+        self._settings = settings
 
-def get_settings(data_id=None):
-    global SETTINGS
 
-    if data_id is None:
-        return _as_dict(SETTINGS)
-    return SETTINGS.get(data_id)
+class DefaultStore:
+    def __init__(self):
+        self._data_store = dict()
+        self._data_names = dict()
 
-
-def get_metadata(data_id=None):
-    global METADATA
-
-    if data_id is None:
-        return _as_dict(METADATA)
-    return METADATA.get(data_id)
+    # Use int for data_id for easier sorting
+    def build_data_id(self):
+        if len(self._data_store) == 0:
+            return 1
+        return max(list(map(lambda x: int(x), self._data_store.keys()))) + 1
+
+    # exposing  _data_store for custom data store plugins.
+    @property
+    def store(self):
+        return self._data_store
+
+    @store.setter
+    def store(self, new_store):
+        self._data_store = new_store
+
+    def keys(self):
+        return list(self._data_store.keys())
+
+    def items(self):
+        return self._data_store.items()
+
+    def contains(self, key):
```

Comment:
Can we add a unit test for this `contains` function?

Reply:
It seems pretty straight forward...You sure?

Reply:
Forget it, if I ever get around to it I'll add one.  Definitely not worth blocking